### ISSUE TITLE
Feedback for ch02

### ISSUE THREAD

**Issue by simoncorstonoliver** [zackchase/mxnet-the-straight-dope#204]
Shouldn't there be an introduction for the chapter?

chapter02_supervised-learning/linear-regression-scratch.ipynb

* Instructions for installing matplotlib but notebooks in ch01 used it without describing how to install. Just put those preqeqs somewhere at the beginning?
* "Train a model means making" >> "Training..."
* Need explanation for this choice: "In this case, we'll use the squared distance between our prediction and the true value."
* This needs elaboration: "It turns out that linear regression actually has a closed-form solution. "

chapter02_supervised-learning/linear-regression-gluon.ipynb

chapter02_supervised-learning/perceptron.ipynb
* Perceptron == E. coli?? I do not see the analogy
* This page is much more math-intensive than what has preceded. Rework to provide text-oriented explanation with subsequent deep dive on math?

chapter02_supervised-learning/softmax-regression-scratch.ipynb
* What is the basis for this claim? Maybe just say it's widely used. "The relevant loss function here is called cross-entropy and it may be the most common loss function you'll find in all of deep learning. That's because at the moment, classification problems tend to be far more abundant than regression problems."
* Inappropriate tone (defined as mild oath by Merriam Websters): Jeepers. 
* Authorial voice of book is "we" in Ch01: "I reviewed"

chapter02_supervised-learning/softmax-regression-gluon.ipynb

* Peculiar turn of phrase: "We won't suck up too much wind"

chapter02_supervised-learning/regularization-scratch.ipynb

* no prior mention in book of decision trees: "decision trees versus neural networks"
* typo: formulizes

chapter02_supervised-learning/loss.ipynb

* Raw math formula: $$\mathop{\mathrm{minimize}}_f \sum_i |y_i - f|$$

chapter02_supervised-learning/environment.ipynb

* Needs elaboration: "he click-through rate for NOKIA phone ads"
* authorial voice: " I had"
* typo: bild
* raw math formula: $$\mathop{\mathrm{minimize}}_w \frac{1}{m} \sum_{i=1}^m l(x_i, y_i, f(x_i)) + \frac{\lambda}{2} \|w\|_2^2$$
* No navigation to the first section of chapter 3

**Comment by zackchase** [comment]
Thanks! This is extremely helpful. I've been making a pass turning sections from sketches into full first drafts (see the introduction in crash course as an example) over the past week. Also doing the hard work of folding contributed sections that I didn't write into the style of the rest of the book. This feedback is clear, precise, and actionable. I hope you have the bandwidth to provide more of it.  👍
